src/storage/store/node_table.cpp

```diff
@@ -313,5 +322,20 @@ uint64_t NodeTable::getEstimatedMemoryUsage() const {
     return nodeGroups->getEstimatedMemoryUsage();
 }
 
+bool NodeTable::isVisible(const transaction::Transaction* transaction, offset_t offset) const {
+    auto [nodeGroupIdx, offsetInGroup] = StorageUtils::getNodeGroupIdxAndOffsetInChunk(offset);
+    auto* nodeGroup = getNodeGroup(nodeGroupIdx);
+    if (nodeGroup->isDeleted(transaction, offsetInGroup)) {
+        return false;
+    }
+    return nodeGroup->isInserted(transaction, offsetInGroup);
+}
+
+bool NodeTable::lookupPK(transaction::Transaction* transaction, common::ValueVector* keyVector,
+    uint64_t vectorPos, common::offset_t& result) const {
```

Comment:
Can u leave a todo here for me? We need to check the transaction local hash index as well.